### PR TITLE
Logistic regression examples

### PR DESCRIPTION
**Pull request by tdoublep**
Added notebooks for logistic regression on the following datasets:
- Avazu
- Allstate (insurance claims)
- Higgs
- Epsilon
- Susy
- Mnist8m

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,52 @@
-# snapml-examples
-IBM Snap ML Examples
+# IBM Snap ML Examples
```
Comment (andreea-anghel) [inline README.md:1]: Do we introduce somewhere the acronym Snap ML?

File: examples/datasets/allstate.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+            p = subprocess.Popen(['unzip', 'train_set.zip'], cwd=self.working_dir)
+            p.wait()
+            
+
+            df = pd.read_csv(os.path.join(self.working_dir, 'train_set.csv'))
+            df.drop(['Row_ID'],axis=1,inplace=True)
+            df.replace('?', np.NaN, inplace=True)
+            df.fillna(-1, inplace=True)
+
+            df_Y = df[['Claim_Amount']]
```
Comment (andreea-anghel) [inline examples/datasets/allstate.py:64]: we need to make sure that we don't reuse code from external sources, e.g., for preprocessing. this comments is valid for all datasets.

File: examples/datasets/mnist8m.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+            p = subprocess.Popen(['wget', 'https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/multiclass/mnist8m.scale.bz2'], cwd=self.working_dir)
+            p.wait()
+
+            p = subprocess.Popen(['bunzip2', 'mnist8m.scale.bz2'], cwd=self.working_dir)
+            p.wait()
+
+            X, y = load_svmlight_file(os.path.join(self.working_dir, "mnist8m.scale"))
+
+            X_train, X_test, y_train, y_test = train_test_split(
```
Comment (andreea-anghel) [inline examples/datasets/mnist8m.py:57]: Is mnist8m multiclass? does train_test_split preserve the class ratios? asking just to see if we can end up with imbalanced datasets.

**Review by andreea-anghel — APPROVED** [review]
looks good to me - subject to my comments

**Comment by andreea-anghel** [comment]
- Avazu: do we have permission to use that image (which is cool by the way ;) )?

File: examples/datasets/allstate.py (hunk 2 above)
Comment (tdoublep) [reply examples/datasets/allstate.py:64]: All of the other datasets are our own code except this one. I asked you whether it came from somewhere else and you pointed me to the publicly available notebook written by the India team.

File: examples/datasets/mnist8m.py (hunk 3 above)
Comment (tdoublep) [reply examples/datasets/mnist8m.py:57]: I've no idea, this is the same dataset that we used for the examples in the PowerAI distribution.

File: examples/datasets/allstate.py (hunk 2 above)
Comment (andreea-anghel) [reply examples/datasets/allstate.py:64]: Then I guess we're safe. That code is under IBM license so it's fine, we can reuse it.

File: examples/datasets/mnist8m.py (hunk 3 above)
Comment (andreea-anghel) [reply examples/datasets/mnist8m.py:57]: Yes, it's multi-class, we used it for testing multi-class Logistic Regression. 
That's fine, the accuracy makes sense in its notebook example 

`https://github.com/IBM/snapml-examples/blob/tpa-start/examples/training/logistic_regression/example_mnist8m.ipynb`

**Comment by tdoublep** [comment]
> * Avazu: do we have permission to use that image (which is cool by the way ;) )?

It is on the competition website and it's just a link to it, not sure if that's a problem

File: README.md (hunk 1 above)
Comment (tdoublep) [reply README.md:1]: added it
